File: src/download.py
import subprocess
import json

def run_command(command):
    p = subprocess.Popen(command,
                         stdout=subprocess.PIPE,
                         stderr=subprocess.STDOUT)
    body = ''
    for line in iter(p.stdout.readline, b''):
        body += line
    return body

data = {}
for subreddit in ['dataisbeautiful', 'earthporn', 'Seattle', 'birdswitharms' ,'cityporn', 'pics', 'programmerhumor']:
    json_data = run_command(['bash', 'download.sh', subreddit])
    data[subreddit] = json.loads(json_data)

print json.dumps(data)

File: src/download.js
var http = require('http');

var subreddits = ['dataisbeautiful', 'earthporn'];

subreddits.forEach(function(subreddit) {
	var data = http.get('http://www.reddit.com/r/' + subreddit + '.json?limit=100', function(res) {
	  	var body = '';
	  	res.on('data', function(chunk){
	        body += chunk;
	    });

	    res.on('end', function() {
	        var data = JSON.parse(body);
	        console.log("Got a response: ", data);
	    });
  	});
});
